### PR TITLE
Add CLI command for Reset activity API

### PR DESCRIPTION
**Pull request by ychebotarev**
## What was changed

Add CLI support for Reset activity API

## Why?

Part of the activity API work.

### CODE REVIEW

File: temporalcli/commands.activity_test.go [hunk 1]
```diff
@@ -231,6 +231,37 @@ func (s *SharedServerSuite) TestActivityUnPause_Failed() {
...
+	res := s.Execute(
+		"activity", "reset",
+		"--activity-id", aid,
+		"--workflow-id", wid,
+		"--run-id", run.GetRunID(),
+		"--identity", identity,
+		"--address", s.Address(),
+	)
+
+	s.NoError(res.Err)
```
Comment (cretz) [inline temporalcli/commands.activity_test.go:249]: Didn't mention it on the last review but I should have. Ideally these tests should confirm the invoked action actually occurs.

**Comment by josh-berry** [comment]
@ychebotarev Do you think you'll be able to get this merged soon? Would like to cut the next CLI release soonish (once a couple other things land) and would love to take this too if I can.

File: temporalcli/commands.activity_test.go (hunk 1 above)
Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:249]: We don't return any values for that. No error indicates that the command was processed successfully. I don't want to write more complicated test - it will be a duplicate of existing functional tests.

Comment (cretz) [reply temporalcli/commands.activity_test.go:249]: How do you know this command did anything? Sometimes we do replicate functional tests in every place we do them. When these tests are written in each SDK (and the CLI is basically an SDK), we make sure the intended effect occurs, yes even if that technically is duplicated in each language.

Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:249]: I know that this command reach the server and return no errors. Which means it was a valid call, and it was processed. 
How do I know it was correctly processed? Functional tests. 
I don't want to replicate those functional tests here,  don't want to test server functionality in CLI tests.

Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:249]: > Not all of them, just confirm it did something

>>I know that this command reach the server and return no errors. 
That is enough.

Comment (josh-berry) [reply temporalcli/commands.activity_test.go:249]: I think Chad's point is that this test isn't actually testing anything. Any implementation of `activity reset` within the CLI, including no implementation at all, would allow this test to pass. We need to check some observable side-effect that occurs as a result of this call.

That doesn't mean we have to replicate all the functional tests the server does. But it does mean checking that the user would see some expected side-effect (any one will do) as a result of running this command.

Hope that clarifies what's expected here.

Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:249]: Unlike for example for updating activity there is no direct expected side effect - the response is empty. 
If there will be any kind of error on server side - like "Not Found" - it will return an error.

Even If the response was not empty - one may apply the same argument  "what if I just return expected result directly without calling OSS".
So the proper test will be:
* make sure that OSS call happens
* independently make sure that expected effect take place

In this case (just like in my functional test) for the proper testing I need to make activity fail few times, then make DescribeWorkflow call and verify that the number of attempts is greater then 1.
Then block it (otherwise test becomes non-deterministic), then make CLI reset activity call, then make DescribeWorkflow call and verify that the number of attempts is 1. This is literally functional test, just much harder.

Comment (cretz) [reply temporalcli/commands.activity_test.go:249]: Yes, it'd basically be a functional test that resets after attempt > 1 and confirms it goes back to 1. You don't have to provide full coverage like you do where the implementation is, but at least one end-to-end assertion is worth it. We will surely be writing this same functional test for every SDK we provide this high-level activity reset call on.

Comment (ychebotarev) [reply temporalcli/commands.activity_test.go:249]: >Yes, it'd basically be a functional test
And that is why  there is no need to do that.

Comment (cretz) [reply temporalcli/commands.activity_test.go:249]: There is a need and we do these kinds of end-to-end tests in other tests in all of our SDKs (which includes the CLI). Right now if the CLI code did nothing, this assertion would pass. Or if it did something but never actually made the server invocation, you'd never know. Even with the next test testing for failure, you don't even know what failure it might hit.

**Review by cretz — APPROVED** [review]
